File: Assets/Scripts/StartMenu/RegistrationService.js
class RegistrationService {
    public function sendRegistrationData(registrationData : RegistrationData, responseHandler : function(Response)) {
        var response : Response = new Response(); 
        
    	var cs = GameObject.Find("GBaaSObject");
    	var script : GBaaSObject;
    	script = cs.GetComponent("GBaaSObject");
    	script.Init();
	
    	var result = script.CreateUser(registrationData.username, registrationData.password, registrationData.email);
    	
    	yield result; // StartCoroutine must need yield ~!!!
    	
    	if(result.length > 0) {
            // Registration was successful:
            Debug.Log("You was registered successfully");
            response.error = false;
            response.message = "";
        } else {
            Debug.Log("Error: Registration to Server");
            response.error = true;
            response.message = "Error: Registration to Server";
        }
        
        // Calling response handler:
        responseHandler(response);
    }
}

File: Assets/Scripts/StartMenu/RegistrationService.cs
﻿using UnityEngine;

using GBaaS.io;
using GBaaS.io.Objects;

interface RegistrationServiceListener {
	void	registrationResponseHandler(Response response);
}

class RegistrationService : GBaaSApiHandler {
	private RegistrationServiceListener _listener = null;
	
	public void SetListener(RegistrationServiceListener listener) {
		_listener = listener;
	}
	
	public void sendRegistrationData(RegistrationData registrationData) {
		Debug.Log("sendRegistrationData " + registrationData.username);
		GBaaSObject.Instance.Init(this);
		GBaaSObject.Instance.CreateUser(registrationData.username, registrationData.name, registrationData.password, registrationData.email);
    }
	
	public override void OnCreateUser(GBResult result) {
		Debug.Log("OnCreateUser " + result.reason);

		Response response = new Response(); 

		if(result.isSuccess) {
			// Registration was successful:
			Debug.Log("You was registered successfully");
			response.error = false;
			response.message = "";
		} else {
			Debug.Log("Error: Registration to Server");
			response.error = true;
			response.message = "Error: Registration to Server";
		}

		// Calling response handler:
		_listener.registrationResponseHandler(response);
	}
}
